tests/queries/0_stateless/02420_force_select_final_setting.sql

```diff
@@ -0,0 +1,48 @@
+-- { echoOn }
```

Comment:
Check it also with EXPLAIN SYNTAX
and with enabled Analyzer as well.

If the Analyzer supports FINAL in JOIN as it should, let's add this case as well - when only the left table or only the right table or both support the FINAL.

There should be a test case for subqueries.

Reply:
I have added a test with explain syntax. Analyzer does not seem to work with `set final = 1`, need to investigate why

Reply:
I have implemented a Pass (`DB::AutoFinalOnQueryPass`) that makes it work with analyzer enabled